packages/web-app/src/reducers/EntranceReducer.js

```diff
@@ -33,6 +33,12 @@ const initialState = {
   latestHttpCode: null
 };
 
+function arrFindReplaceOrAdd(arr, test, replacement) {
```

Comment:
Let's not mutate the redux store

[redux doc](https://redux.js.org/usage/structuring-reducers/immutable-update-patterns#inserting-and-removing-items-in-arrays)

If necessary, we can get inspiration from redux examples or use a small dependency like [immer](https://github.com/immerjs/immer)